### DISCUSSION TITLE
Argo-CD changelog is not updated since July 2022

### DISCUSSION THREAD

**Discussion by ramanNarasimhan77** [argoproj/argo-cd#13561] [General]
I was trying to compare two ArgoCD release versions using the Changelog to determine if I really should upgrade, but surprisingly I found that Changelog https://github.com/argoproj/argo-cd/blob/master/CHANGELOG.md was last updated 9 months ago. It does not have information after release v2.4.8 (2022-07-29)

**Comment by asfaltboy** [comment]
I personally find a changelog quite useful as it allows me to easily find a change or review all changes from one version to another in one simple view.

I have noticed, though, that the changes are still recorded in each released version, under [GitHub releases](https://github.com/argoproj/argo-cd/releases).

The main use case of changelog is to help figure out what changed from version X to version Y. This is mostly done before upgrading, or when debugging a bug or a regression.
From this perspective, GitHub's release comparison doesn't help much as it results in a commit log ([for example 2.5.16 to 2.7.15](https://github.com/argoproj/argo-cd/compare/v2.7.2...v2.5.16)), which includes administrative commits and can be harder to grok than the human-readable changelog format.
To make things more complicated, there are also guides for upgrading from one "minor" or "major" version to the next, which should be consulted when considering an upgrade.

Could it be that at some point in time, contributors decided it's no longer necessary to update the CHANGELOG.md file but never followed up with cleanup?

If this is the case, I imagine we'd want to:
1. Update the instructions for updating changelog in [the docs](https://argo-cd.readthedocs.io/en/latest/developer-guide/releasing/), with the new process that uses release notes
2. Explain where release notes are found and how to diff two versions in a dedicated docs page (possibly using a tool like `gh`.
3. Add a link to said release notes docs page in the GitHub README, the upgrade page, and others, for easy access.
4. Update CHANGELOG.md to clarify it is now out of date, and point to the same doc above.
5. Consider adding a step-by-step guide as a recommended read before upgrading, to reduce further toil from users.
